src/query/service/src/interpreters/interpreter_copy.rs

```diff
@@ -254,67 +240,120 @@ impl CopyInterpreter {
                     .files_to_copy
                     .clone()
-                    .ok_or(ErrorCode::Internal("files_to_copy should not be None"))?,
-            )
-        } else {
-            let table_ctx: Arc<dyn TableContext> = self.ctx.clone();
-            let files = plan.collect_files(&table_ctx).await?;
-            let mut build_res = PipelineBuildResult::create();
-            if files.is_empty() {
-                return Ok(build_res);
+                    .ok_or(ErrorCode::Internal("files_to_copy should not be None"))?;
+
+                let (query_build_res, query_source_schema) = self.build_query(query).await?;
+                build_res = query_build_res;
+                source_schema = query_source_schema;
             }
-            self.build_read_stage_table_data_pipeline(
-                &mut build_res.main_pipeline,
-                plan,
-                to_table.get_block_thresholds(),
-                files.clone(),
-            )
-            .await?;
-            (build_res, plan.required_source_schema.clone(), files)
-        };
+        }
 
-        build_append_data_pipeline(
-            ctx.clone(),
-            &mut build_res.main_pipeline,
-            CopyPlanType::CopyIntoTablePlanOption(plan.clone()),
-            source_schema,
-            to_table.clone(),
-        )?;
+        let file_sizes: u64 = files.iter().map(|f| f.size).sum();
 
-        // if it's replace mode, don't commit, because COPY is the source of replace.
-        match plan.write_mode {
-            CopyIntoTableMode::Replace => set_copy_on_finished(
-                ctx,
-                files,
-                plan.stage_table_info.stage_info.copy_options.purge,
-                plan.stage_table_info.stage_info.clone(),
+        // Perf
+        {
+            metrics_inc_copy_read_file_size_bytes(file_sizes as u32);
+            metrics_inc_copy_read_file_counter(files.len() as u32);
+        }
+
+        // Append data.
+        {
+            self.set_status(&format!(
+                "Copy begin to append data: {} files, size_in_bytes:{} into table",
+                files.len(),
+                file_sizes
+            ));
+
+            let start = Instant::now();
+            build_append_data_pipeline(
+                ctx.clone(),
                 &mut build_res.main_pipeline,
-            )?,
-            _ => {
-                // commit.
-                build_commit_data_pipeline(
-                    ctx.clone(),
-                    &mut build_res.main_pipeline,
-                    plan.stage_table_info.stage_info.clone(),
-                    to_table,
+                CopyPlanType::CopyIntoTablePlanOption(plan.clone()),
+                source_schema,
+                to_table.clone(),
+            )?;
+
+            // Perf
+            {
+                metrics_inc_copy_read_file_cost_milliseconds(start.elapsed().as_millis() as u32);
+                self.set_status(&format!(
```

Comment:
why do we need to metric and status for  `build_append_data_pipeline`?
it does not really copy any data, just builds the pipeline.

Reply:
How can i measure the copy data cost?

Reply:
we can get time for 
 
1. special steps when bind and get_partitons
1. the whole pipeline execution time
2. inside each processor for each part/block.

I can refine this in another PR.